File: AuthServer/AuthServer.Application/UseCases/CreateMedia/MediaTypeExtractor.cs
using Media.Models;
using Microsoft.AspNetCore.Http;

namespace AuthServer.Application.UseCases.CreateMedia
{
    internal class MediaTypeExtractor
    {
        public MediaType Extract(IFormFile media)
        {
            if (!media.ContentType.StartsWith("image"))
                throw new Exception("Invalid media type.");

            return MediaType.Image;
        }
    }
}

File: AngularClient/app_angular_client/src/app/media/services/media-type-extractor.ts
import { Injectable } from "@angular/core";
import { InvalidMediaTypeError } from "../errors/invalid-media-type-error";
import { MediaType } from "../media-type";

@Injectable({
  providedIn: 'root'
})
export class MediaTypeExtractor{

  extract(file: File) : MediaType{
    if(file.type.startsWith("image/"))
      return MediaType.image;
    else if(file.type.startsWith("video/"))
      return MediaType.video;
    throw new InvalidMediaTypeError();
  }

}
